lib/diagnostic.js

```diff
@@ -6,29 +6,55 @@
 // splits it into an array of normalized words
 // (uppercase strings without punctuation)
 // and returns that array.
-const normalizeWords = function (/* */) {
-  /* your response here */
+const normalizeWords = function (paragraph) {
+  let words = paragraph.split(/\s+/);
+  for (let i = 0; i < words.length; i++) {
+  let word = words[i].replace(/\W+/, '');
+  normalizeWords[i] = word.toUpperCase();
+}
 };
```

Comment:
Really close! A few things:
1) be careful the names of the variables/functions you are calling. In line 13, you are calling `normalizeWords` inside of the definition of `normalizeWords`. That makes this a "recursive" function, which is probably not what you intended. 
2) the thing you're missing is a `return` statement, outside of the for-loop, that returns the list of words that have been capitalized